x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -485,6 +498,7 @@ class ProcessContext {
                 PROCESS_NAME,
                 task.getParams().getQueueCapacity()
             );
+            this.startsCount.set(startsCount == null ? 1 : startsCount);
```

Comment:
nit: `startsCount` can be an `int` set to 1 the first time it is called in startDeployment() line 138. Save a null check.